docs/modules/guides/pages/pipeline-with-proxy.adoc

```diff
@@ -0,0 +1,94 @@
+= Jenkins pipeline using browser proxy in Docker container
+
+```groovy
+def vividusNetwork = 'vividus-grid'<1>
+def vividusHub = 'vividus-hub'<2>
+def vividusBrowser = 'vividus-browser'<3>
+def vividusProxyHost = 'vividus-proxy-host'<4>
+def seleniumVersion = '3.141.59-20210804'<5>
+def overridings = """
+                        proxy.host=${vividusProxyHost}<4>
+                        proxy.connectable-host=${vividusProxyHost}<4>
+                        selenium.grid.enabled=true
+                        selenium.grid.url=http://${vividusHub}:4444/wd/hub<2>
+                        selenium.grid.capabilities.platformName=LINUX
+                  """
+projectName = 'vividus-test-project'
+pipeline {
+  agent {
+    label 'docker'
+  }
+
+  stages {
+    stage('Start Selenium Grid') {
+        steps {
+           sh "docker network create ${vividusNetwork}"<1>
+           sh "docker run -d --net ${vividusNetwork} --name ${vividusHub} selenium/hub:${seleniumVersion}"<2>
+           sh "docker run -d --net ${vividusNetwork} -e HUB_HOST=${vividusHub} -e SCREEN_WIDTH=1920 -e SCREEN_HEIGHT=1080 --shm-size=2g --name ${vividusBrowser} selenium/node-chrome:${seleniumVersion}"<3><5>
+        }
+    }
+
+    stage('Load Source') {
+        steps {
+            checkout([$class: 'GitSCM',
+                branches: [[name: master]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [
+                    [$class: 'RelativeTargetDirectory', relativeTargetDir: projectName],
+                    [$class: 'SubmoduleOption',
+                    disableSubmodules: false,
+                    parentCredentials: true,
+                    recursiveSubmodules: true,
+                    reference: '',
+                    trackingSubmodules: false]
+                ],
+                submoduleCfg: [],
+                userRemoteConfigs: [[credentialsId: scmCredentialId, url: gitUrl]]])
```

Comment:
how does it work? I am curious about `name: master`, `relativeTargetDir: projectName`, `credentialsId: scmCredentialId, url: gitUrl`